Products/CMFCore/MemberDataTool.py

```diff
@@ -235,11 +235,12 @@ def __init__(self, user, tool):
         self._tool = tool
         self.__parent__ = aq_parent(aq_inner(user))
         id = user.getId()
-        self._md = tool._members.setdefault(id, MemberData(id))
+        self._md = tool._members.get(id, MemberData(id))
 
     @security.private
     def notifyModified(self):
         # Links self to parent for full persistence.
+        self._tool._members[id] = self._md
```

Comment:
`id` is not defined here. It triggers no warnings because `id` is a builtin.

Reply:
`s/id/self._user.getId()/`.